src/authenticated-app.js

```diff
@@ -31,7 +36,28 @@ function ErrorFallback({ error }) {
 }
 
 function AuthenticatedApp() {
-  const { user, logout } = useAuth();
+  const notify = () => { toast.success('Login successfully!'); };
+  const navigate = useNavigate();
+  const location = useLocation();
+  React.useEffect(() => {
+    if (location.pathname === '/authentication') {
+      navigate('/discover');
+      notify();
+    }
+  }, [location.pathname, navigate]);
+  const { user } = useSelector(selectUser);
+  const dispatch = useDispatch();
+  const handleLogout = () => {
+    navigate('/authentication');
+    dispatch(logout());
+    dispatch(resetBook());
+    dispatch(resetListItem());
```

Comment:
Have you tried to do the research that if we can trigger only 1 action `logout`, then this action can update state in multiple reducers?